tests/engine/test_launch.py

```diff
@@ -141,6 +141,24 @@ def test_submit_store_provenance_false(self):
         with self.assertRaises(exceptions.InvalidOperation):
             launch.submit(AddWorkChain, term_a=self.term_a, term_b=self.term_b, metadata={'store_provenance': False})
 
+    def test_submit_process_function(self):
+        """Test the `submit` of process function."""
+        inputs = {'term_a': orm.Int(2), 'term_b': orm.Int(4)}
+        node = launch.submit(add, **inputs)
+        self.assertIsInstance(node, orm.CalcFunctionNode)
```

Comment:
This merely tests that the `submit` method does not except and returns `CalcFunctionNode`, which is a good start, but we should also test that the process can actually be executed by a daemon runner. You can try to add this test here, but you will need to fake the daemon runner of course. You can try and take inspiration of the `tests/cmdline/commands/test_process.py` tests, or you can add submitting a process function to the `.ci/test_daemon.py` script. This will also run in the CI but with a normal daemon running, making it easier to test this kind of stuff.  You should test submitting a process function that is defined normally and is importable through the `PYTHONPATH` as well as through an entry point. For the latter you can use `aiida.workflows:arithmetic.add_multiply` which comes with `aiida-core` so you don't have to create one especially for the test.

Reply:
Thanks! I add tests in `.ci/test_daemon.py`. That's true these 'real' tests bring out a subtle issue, that `recreate_from` method should be given to the process function.